nacos/src/main/java/com/linecorp/armeria/internal/nacos/NacosClientBuilder.java

```diff
@@ -0,0 +1,126 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.nacos;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.nacos.NacosConfigSetters;
+
+public final class NacosClientBuilder implements NacosConfigSetters<NacosClientBuilder> {
+    private final URI nacosUri;
+    private final String serviceName;
+
+    private String nacosApiVersion = DEFAULT_NACOS_API_VERSION;
+
+    @Nullable
+    private String username;
+
+    @Nullable
+    private String password;
+
+    @Nullable
+    private String namespaceId;
+
+    @Nullable
+    private String groupName;
+
+    @Nullable
+    private String clusterName;
+
+    @Nullable
+    private Boolean healthyOnly;
+
+    @Nullable
+    private String app;
+
+    NacosClientBuilder(URI nacosUri, String serviceName) {
+        this.nacosUri = requireNonNull(nacosUri, "nacosUri");
+        this.serviceName = requireNonNull(serviceName, "serviceName");
+    }
+
+    @Override
+    public NacosClientBuilder namespaceId(String namespaceId) {
+        this.namespaceId = requireNonNull(namespaceId);
+
+        return this;
+    }
+
+    @Override
+    public NacosClientBuilder groupName(String groupName) {
+        this.groupName = requireNonNull(groupName);
+
+        return this;
+    }
+
+    @Override
+    public NacosClientBuilder clusterName(String clusterName) {
+        this.clusterName = requireNonNull(clusterName);
+
+        return this;
+    }
+
+    @Override
+    public NacosClientBuilder app(String app) {
+        this.app = requireNonNull(app);
+
+        return this;
+    }
+
+    @Override
+    public NacosClientBuilder nacosApiVersion(String nacosApiVersion) {
+        checkArgument(NACOS_API_VERSION_PATTERN.matcher(nacosApiVersion).matches(),
+                      "nacosApiVersion: %s (expected: a version string that starts with 'v', e.g. 'v1')",
+                      nacosApiVersion);
+        this.nacosApiVersion = requireNonNull(nacosApiVersion, "nacosApiVersion");
+
+        return this;
+    }
+
+    @Override
+    public NacosClientBuilder authorization(String username, String password) {
+        requireNonNull(username, "username");
+        requireNonNull(password, "password");
+
+        this.username = username;
+        this.password = password;
+
+        return this;
+    }
+
+    public NacosClientBuilder healthyOnly(Boolean healthyOnly) {
+        this.healthyOnly = requireNonNull(healthyOnly);
+
+        return this;
+    }
+
+    public NacosClient build() {
+        final URI uri;
+        try {
+            uri = new URI(nacosUri.getScheme(), null, nacosUri.getHost(), nacosUri.getPort(),
+                          "/nacos", null, null);
```

Comment:
@minwoox 

While /nacos is the default path, but it could be behind a proxy and there is no guarantee that it cannot be changed through configuration. Therefore, it is better to allow users to configure it directly. 

I'll remove it. Thanks.